sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -129,6 +129,7 @@ package object util extends Logging {
     case a: Attribute => new PrettyAttribute(a)
     case Literal(s: UTF8String, StringType) => PrettyAttribute(s.toString, StringType)
     case Literal(v, t: NumericType) if v != null => PrettyAttribute(v.toString, t)
+    case Literal(null, t) => PrettyAttribute("NULL", t)
```

Comment:
Seems this is an orthorhombic issue, I'm not sure if handle it in this PR or separate it. @cloud-fan

Reply:
Let's open a new PR to fix it.

Reply:
Created [#31233](https://github.com/apache/spark/pull/31233)